app/background/setting/service.js

```diff
@@ -37,6 +37,21 @@ export async function setCustomLocale(json) {
   return await put(CUSTOM_LOCALE, JSON.stringify(json));
 }
 
+export async function getLatestRelease() {
+  try {
+    const releases = await (
+      await fetch(
+        'https://api.github.com/repos/kyokan/bob-wallet/releases'
```

Comment:
Mine as well check some kind of signature, even if its a brand new bob-release-update published by kyokan. Especially because you use `url: latestRelease.html_url,` down below. Giving users a link to click is a target for attacks, treat it like a nuclaar bomb.

https://bitcoinist.com/electrum-wallet-phishing-bitcoin/

Reply:
Wow didn't think of that. Unlikely, but def possible.

How about always linking to `https://github.com/kyokan/bob-wallet/releases/tag/{TAG}` instead of `latestRelease.html_url`?

Not sure what signature can be checked here. Do you mean like with PGP? It's just a link so no info to verify.

Reply:
Updated to prefix url and only use tag. imo the only way to break this would be to MITM a HTTPS req to github.com, which if someone can do, there are bigger problems already.

>maybe an HNS address so we can use rpc verify message

Requires the node to be synced :|

>I dunno if it's possible to check that the release signature on github is correct before notifying the user.

Afaik, no. We could download another file and check that, but again relies on an external source, mostly over https, with the same attack vectors as https://github.com. Additionally, verifying a file would mean downloading the actual binary from within Bob which feels more dangerous.

Reply:
> Requires the node to be synced :|

no, just the address, not the NAME.